students/andrew_garcia/lesson_04/Mailroom Part 2.py

```diff
@@ -0,0 +1,167 @@
+        filename = item[0] + '-' + str(item[1])  # creates file named after donor and donation amount
+        with open(f'{filename}.txt', 'w+') as f:
+            text = (f'''\nDear {item[0]},
+    
+            Thank you so much for your generous donation of ${item[1]:.2f}!
+            The kind generosity that you and others give us helps us complete the great work we are doing for others.
+    
+            From,
+            Your Local Charity
+            \n''')
+            f.write(text)
+    print('Thank You Saved.')
+    options_menu()
+
+def format_thank_you(name, amount):
+    """Creates thank you note for one donor"""
+
+    print(f'''\nDear {name.title()},
+
+Thank you so much for your generous donation of ${amount:.2f}!
+The kind generosity that you and others give us helps us complete the great work we are doing for others.
+
+From,
+Your Local Charity
+\n''')
+    options_menu()
+
+def existing_donor(name, all_donations = all_donations):
+    """Creates a Thank You Letter based on existing donations"""
+
+    if name.title() in all_donations:  # looks for donation amounts from a donor
+        print('List of Donations:')
+        for item in all_donations[name.title()]:
+            print(item)
+
+        donation_number = float(input('Which donation amount would you like to use?: '))
+        if donation_number in all_donations[name.title()]:  # takes a donation amount anf formats a thank you
+            format_thank_you(name, donation_number)
+        else:
+            print('That is not a valid donation amount!')
+            options_menu()
+
+def add_donation(name, all_donations = all_donations):
+    """Takes a new or existing donor and adds their donation to the list"""
+
+    amount = input(f'How much did {name} donate?: ')
+    if name.title() in all_donations:  # adds donation amount to dictionary with existing donor
+        all_donations[name.title()] += [amount]
+    elif name.title() not in all_donations:  # adds donor to the dictionary if it does not exist
+        all_donations[name.title()] = [float(amount)]
+    format_thank_you(name, float(amount))
+
+
+def send_thank_you(all_donations = all_donations): #edit for dict
+    """Gets donor name so that a thank you letter can be formatted"""
+
+    name = input('Which Donor would you like to send a note to?: ')
+    if name.lower() == 'list':  # gives list of donors
+        print(all_donations)
+        send_thank_you()
+    elif name.lower() == 'quit':
+        options_menu()
+    elif name.title() in all_donations.keys():  # uses existing donor
+        new_or_old = input('{} is already an existing donor. Would you like to use a new or existing donation?: [new/existing] '.format(name))
+        if new_or_old.lower() == 'new':  # allows user to add a new donation for existing donor
+            add_donation(name)
+        elif new_or_old.lower() == 'existing':  # uses an existing donation amount
+            existing_donor(name)
```

Comment:
can you explain above logic?

wouldn't new vs existing donor be based on whether it exists in the dictionary?